tests/components/yessssms/test_notify.py

```diff
@@ -151,210 +149,219 @@ async def test_connection_error_on_init(hass, caplog, valid_settings, connection
-
-        with self.assertLogs(
-            "homeassistant.components.yessssms.notify", level="ERROR"
-        ) as context:
-            self.yessssms.send_message(message)
-
-        self.assertIn(
-            "You need to provide a recipient for SMS notification", context.output[0]
-        )
-
-    @requests_mock.Mocker()
-    def test_sms_sending_error(self, mock):
-        """Test sms sending error."""
-        mock.register_uri(
-            "POST",
-            # pylint: disable=protected-access
-            self.yessssms.yesss._login_url,
-            status_code=302,
-            # pylint: disable=protected-access
-            headers={"location": self.yessssms.yesss._kontomanager},
-        )
+        yessssms.yesss._websms_url,
+        status_code=200,
+        text="<h1>Ihre SMS wurde erfolgreich verschickt!</h1>",
+    )
+    requests_mock.get(
         # pylint: disable=protected-access
-        login = self.yessssms.yesss._logindata["login_rufnummer"]
-        mock.register_uri(
-            "GET",
-            # pylint: disable=protected-access
-            self.yessssms.yesss._kontomanager,
-            status_code=200,
-            text=f"test...{login}</a>",
-        )
-        mock.register_uri(
-            "POST",
-            # pylint: disable=protected-access
-            self.yessssms.yesss._websms_url,
-            status_code=HTTP_INTERNAL_SERVER_ERROR,
-        )
-
-        message = "Testing YesssSMS platform :)"
-
-        with self.assertLogs(
-            "homeassistant.components.yessssms.notify", level="ERROR"
-        ) as context:
-            self.yessssms.send_message(message)
-
-        self.assertTrue(mock.called)
-        self.assertEqual(mock.call_count, 3)
-        self.assertIn("YesssSMS: error sending SMS", context.output[0])
-
-    @requests_mock.Mocker()
-    def test_connection_error(self, mock):
-        """Test connection error."""
-        mock.register_uri(
-            "POST",
-            # pylint: disable=protected-access
-            self.yessssms.yesss._login_url,
-            exc=yessssms.YesssSMS.ConnectionError,
-        )
-
-        message = "Testing YesssSMS platform :)"
-
-        with self.assertLogs(
-            "homeassistant.components.yessssms.notify", level="ERROR"
-        ) as context:
-            self.yessssms.send_message(message)
-
-        self.assertTrue(mock.called)
-        self.assertEqual(mock.call_count, 1)
-        self.assertIn("cannot connect to provider", context.output[0])
+        yessssms.yesss._logout_url,
+        status_code=200,
+    )
+
+    with caplog.at_level(logging.INFO):
+        yessssms.send_message(message)
+    for record in caplog.records:
+        if (
+            record.levelname == "INFO"
+            and record.name == "homeassistant.components.yessssms.notify"
+        ):
+            assert "SMS sent" in record.message
+
+    assert requests_mock.called is True
+    assert requests_mock.call_count == 4
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)